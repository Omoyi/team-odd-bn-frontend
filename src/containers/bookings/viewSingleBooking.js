import React, { Component } from 'react'
import '../../assets/css/modal.scss'
import { connect } from 'react-redux'
import SubmitButton from '../../components/SubmitButton'
import { openSingleBooking } from '../../redux/actions/bookings/viewSingleBookingActions'

import matchCitiesWithIds from '../../helpers/matchCitiesWithIds'

export class ViewSingleBooking extends Component {

  state={
    modall: true
  }

  handleSubmit  = (e, id, index) => {     
    e.preventDefault();
    this.setState(prevState=>(
      {
        modall: !prevState.modall,
        index
      }
    ))
    this.props.openSingleBooking(true);
  }
  closeModal = props => {
    props.openSingleBooking(false)
  }
  render() {
    
    const {modal, allBookings,handleUpdateModal } = this.props;

    const userBooking = allBookings && allBookings.data.myBookingsresult[this.props.myIndex];
    
    return (
      <>
{    this.state.modall &&  <div id="myModal" className="modal">
        <div className="modal-content modal-content-small">
          <span
            data-test="close-modal"
            role="button"
            tabIndex={0}
            className="close"
            onClick={() =>handleUpdateModal(modal)}
         
          >
            &times;
          </span>
          <h2 id="modal-header" className="modal-header">
            Booking
          </h2>
          <div className="modal-body">
            <div className="center-element">
              <form
                data-test="close-modal-btn"
                onSubmit={() =>handleUpdateModal(modal)}
              >
                <div className="modal-disabled-input">
                  <label>Accommodation</label>
                  <input
                    className="transparent-input"
                    type="text"
                    value={
                      (userBooking && userBooking.room.accommodation.name) || ''
                    }
                    disabled
                  />
                </div>
                <div className="modal-disabled-input">
                  <label>Address</label>
                  <input
                    className="transparent-input"
                    type="text"
                    value={
                      (userBooking && userBooking.room.accommodation.address) || ''
                    }
                    disabled
                  />
                </div>
                <div className="modal-disabled-input">
                  <label>Room type</label>
                  <input
                    className="transparent-input"
                    type="text"
                    value={(userBooking && userBooking.room.roomType) || ''}
                    disabled
                  />
                </div>
                <div className="modal-disabled-input">
                  <label>Room Name</label>
                  <input
                    className="transparent-input"
                    type="text"
                    value={(userBooking && userBooking.room.name) || ''}
                    disabled
                  />
                </div>
                <div className="modal-disabled-input">
                  <label>Check-in date</label>
                  <input
                    className="transparent-input"
                    type="text"
                    value={
                      (userBooking && userBooking.checkInDate.slice(0, 10)) || ''
                    }
                    disabled
                  />
                </div>
                <div className="modal-disabled-input">
                  <label>Check-out date</label>
                  <input
                    className="transparent-input"
                    type="text"
                    value={
                      (userBooking && userBooking.checkOutDate.slice(0, 10)) || ''
                    }
                    disabled
                  />
                </div>
                <SubmitButton buttonName="Close" className="btn-long" />
              </form>
            </div>
          </div>
        </div>
      </div>}
      </>
    )
  }
}

export const mapStateToProps = state => {
  return {
    bookings: state.bookings,
    data: state.bookings.modal.payload && state.bookings.modal.payload.data,
    cities: state.accommodation.createAccommodation.retrievedCities.data,
    allBookings: state.bookings.bookings.allBookings,
  }
}
  
export const mapDispatchToProps = {
  
};


export default connect(mapStateToProps, { openSingleBooking })(ViewSingleBooking)