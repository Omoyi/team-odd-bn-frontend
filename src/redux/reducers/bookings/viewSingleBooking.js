import {
  OPEN_SINGLE_BOOKING, GET_SINGLE_BOOKINGS_SUCCESS, GET_SINGLE_BOOKINGS_ERROR,
} from '../../actionTypes/bookingActionTypes';

const initialState = {
  openSingleBooking: false,
};

const openSingleBookingReducer = (state = initialState, action) => {
  switch (action.type) {
    case OPEN_SINGLE_BOOKING:
      return {
        openSingleBooking: action.payload,
      };
    case GET_SINGLE_BOOKINGS_SUCCESS:
      return {
        ...state,
        payload: action.payload,
      };
    case GET_SINGLE_BOOKINGS_ERROR:
      return {
        payload: action.payload,
      };

    default:
      return state;
  }
};

export default openSingleBookingReducer;
