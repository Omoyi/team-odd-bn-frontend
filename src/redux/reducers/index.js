import { combineReducers } from 'redux';
import socialAuth from './socialAuthReducer';
import authReducers from './authReducers';
import profileReducers from './ProfileReducers';
import updateProfileReducer from './updateProfileReducer';
import getAllUsersReducer from './getAllUsersReducers';
import signoutReducer from './signoutReducer';
import tripReducers from './tripReducers';
import getAllCitiesReducers from './getAllCitiesReducers';
import accommodationReducer from './accommodation';
import multicityReducer from './multicityReducer';
import adminReducer from './admin/adminReducer';
import viewAccomodationReducer from './accommodation/viewAccomodationReducer';
import PaginationReducer from './PaginationReducer';
import searchReducer from './searchReducers/searchReducers';

import bookingReducer from './bookings';

export default combineReducers({
  viewProfile: profileReducers,
  updateProfile: updateProfileReducer,
  allUsers: getAllUsersReducer,
  socialAuth,
  auth: authReducers,
  signout: signoutReducer,
  trips: tripReducers,
  allCities: getAllCitiesReducers,
  accommodation: accommodationReducer,
  multicityTrip: multicityReducer,
  admin: adminReducer,
  allAccomodation: viewAccomodationReducer,
  pagination: PaginationReducer,
  bookings: bookingReducer,
  search: searchReducer,
});
