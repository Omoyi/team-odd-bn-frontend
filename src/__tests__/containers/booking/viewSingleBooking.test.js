import React from 'react';
import { shallow, mount } from 'enzyme';
import { applyMiddleware, createStore } from 'redux';
import thunk from 'redux-thunk';
import { ViewSingleBooking, mapStateToProps } from '../../../containers/booking/viewSingleBooking';
import { defaultProps, initiaState, tempViewAllBooking } from '../../../__mocks__/booking/booking';
import rootReducer from '../../../redux/reducers/index';

const middlewares = [thunk];
const testStore = (state) => {
  const createStoreWithMiddleware = applyMiddleware(...middlewares)(createStore);
  return createStoreWithMiddleware(rootReducer, state);
};
const setUp = (initialState = {}) => {
  const store = testStore(initialState);
  const wrapper = shallow(<ViewSingleBooking {...defaultProps} store={store} />);
  return wrapper;
};

const emptySetUp = (initialState = {}) => {
  const store = testStore(initialState);
  const wrapper = shallow(<ViewSingleBooking {...tempViewAllBooking} store={store} />);
  return wrapper;
};
describe('View single booking test suite', () => {
  it('Should close modal Successfully', () => {
    const component = setUp(defaultProps);
    const closeModalSpy = jest.spyOn(component.instance(), 'closeModal');
    component.find('[data-test="close-modal"]').simulate('click', {
      preventDefault() {},
    });
    component.find('[data-test="close-modal"]').simulate('KeyDown', {
      preventDefault() {},
    });
    expect(closeModalSpy).toHaveBeenCalled();
  });
  it('Should close modal Successfully', () => {
    const component = setUp(defaultProps);
    const closeModalSpy = jest.spyOn(component.instance(), 'closeModal');
    component.find('[data-test="close-modal-btn"]').simulate('submit', {
      preventDefault() {},
    });
    expect(closeModalSpy).toHaveBeenCalled();
  });
  it('Should close modal Successfully', () => {
    const component = emptySetUp(tempViewAllBooking);
    const closeModalSpy = jest.spyOn(component.instance(), 'closeModal');
    component.find('[data-test="close-modal-btn"]').simulate('submit', {
      preventDefault() {},
    });
    expect(closeModalSpy).toHaveBeenCalled();
  });
  it('should map state to props', () => {
    expect(mapStateToProps(initiaState)).toBeDefined();
  });
});
