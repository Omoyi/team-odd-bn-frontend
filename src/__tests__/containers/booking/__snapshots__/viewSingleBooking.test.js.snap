// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expect to view specific Should view specific booking 1`] = `ShallowWrapper {}`;
